packages/framework/aqueduct/src/request-handlers/requestHandlers.ts

```diff
@@ -61,13 +69,21 @@ export const mountableViewRequestHandler = (
  * @param defaultRootId - optional default root data store ID to pass request in case request is empty.
  */
 export const defaultRouteRequestHandler = (defaultRootId: string) => {
-	return async (request: IRequest, runtime: IContainerRuntime) => {
+	return async (
+		request: IRequest,
+		runtime: IContainerRuntime,
+	): Promise<IResponse | undefined> => {
 		const parser = RequestParser.create(request);
 		if (parser.pathParts.length === 0) {
-			return runtime.IFluidHandleContext.resolveHandle({
-				url: `/${defaultRootId}${parser.query}`,
-				headers: request.headers,
-			});
+			try {
+				return createFluidObjectResponse(
```

Comment:
here is an of how a changed resolveHandle can be adapted into the request ecosystem